tests/balance.rs

```diff
@@ -49,3 +49,187 @@ fn verify_get_balance() {
+        assert_eq!(ini_blnc_dave + AMOUNT, now_blnc_dave);
+    })
+}
+
+#[test]
+fn verify_multiple_transfers_same() {
+    const AMOUNT: u128 = 100;
+
+    let (alice_addr_32, alice_addr_mv) = addrs_from_ss58(ALICE_ADDR).unwrap();
+    let (bob_addr_32, bob_addr_mv) = addrs_from_ss58(BOB_ADDR).unwrap();
+
+    new_test_ext().execute_with(|| {
+        // Check initial state of balances of involved users.
+        let ini_blnc_alice = Balances::free_balance(&alice_addr_32);
+        let ini_blnc_bob = Balances::free_balance(&bob_addr_32);
+
+        // Now check that it works from within the MoveVM.
+        let script = assets::read_script_from_project("balance", "double_transfer");
+
+        let src = bcs::to_bytes(&alice_addr_mv).unwrap();
+        let dst = bcs::to_bytes(&bob_addr_mv).unwrap();
+        let amount = bcs::to_bytes(&AMOUNT).unwrap();
+        let params: Vec<&[u8]> = vec![&src, &dst, &amount, &dst, &amount];
+        let transaction = Transaction {
+            script_bc: script,
+            type_args: Vec::<TypeTag>::new(),
+            args: params.iter().map(|x| x.to_vec()).collect(),
+        };
+        let transaction_bc = bcs::to_bytes(&transaction).unwrap();
+
+        assert_ok!(MoveModule::execute(
+            RuntimeOrigin::signed(alice_addr_32.clone()),
+            transaction_bc,
+            1500,
+        ));
+
+        let now_blnc_alice = Balances::free_balance(&alice_addr_32);
+        let now_blnc_bob = Balances::free_balance(&bob_addr_32);
+        assert_eq!(ini_blnc_alice - AMOUNT * 2, now_blnc_alice);
+        assert_eq!(ini_blnc_bob + AMOUNT * 2, now_blnc_bob);
+    })
+}
+
+#[test]
+fn verify_gas_limit_too_low() {
+    const AMOUNT: u128 = 100;
+
+    let (alice_addr_32, alice_addr_mv) = addrs_from_ss58(ALICE_ADDR).unwrap();
+    let (_, bob_addr_mv) = addrs_from_ss58(BOB_ADDR).unwrap();
+
+    new_test_ext().execute_with(|| {
+        // Now check that it works from within the MoveVM.
+        let script = assets::read_script_from_project("balance", "single_transfer");
+
+        let src = bcs::to_bytes(&alice_addr_mv).unwrap();
+        let dst = bcs::to_bytes(&bob_addr_mv).unwrap();
+        let amount = bcs::to_bytes(&AMOUNT).unwrap();
+        let params: Vec<&[u8]> = vec![&src, &dst, &amount];
+        let transaction = Transaction {
+            script_bc: script,
+            type_args: Vec::<TypeTag>::new(),
+            args: params.iter().map(|x| x.to_vec()).collect(),
+        };
+        let transaction_bc = bcs::to_bytes(&transaction).unwrap();
+
+        assert!(MoveModule::execute(
+            RuntimeOrigin::signed(alice_addr_32.clone()),
+            transaction_bc,
+            99,
+        )
+        .is_err());
+    })
+}
+
+#[test]
+fn verify_insufficient_balance() {
+    const AMOUNT: u128 = 100;
+
+    let (alice_addr_32, alice_addr_mv) = addrs_from_ss58(ALICE_ADDR).unwrap();
+    let (_, bob_addr_mv) = addrs_from_ss58(BOB_ADDR).unwrap();
```

Comment:
There are more benefits when dealing with those variables instead of having it as `static ref` like in the mock.rs... At some function calls, the compiler will complain about to implement `sp_runtime::Borrow` for them... but in case of this function use, you don`t have to...